@mkevins Heads up that I’ve started reviewing the chain of PRs. The code LGTM but I’m a bit stuck on testing since I’m facing a startup crash in the WP app (clean install).

```
    java.lang.RuntimeException: Unable to start activity ComponentInfo{org.wordpress.android.beta/org.wordpress.android.ui.main.WPMainActivity}: org.wordpress.android.util.signature.SignatureNotFoundException
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3676)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3813)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:101)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2308)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.app.ActivityThread.main(ActivityThread.java:7898)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
     Caused by: org.wordpress.android.util.signature.SignatureNotFoundException
        at org.wordpress.android.util.signature.SignatureUtils.checkSignatureHashAfterApi28(SignatureUtils.kt:44)
        at org.wordpress.android.util.signature.SignatureUtils.checkSignatureHash(SignatureUtils.kt:20)
        at org.wordpress.android.util.publicdata.ClientVerification.canTrust(ClientVerification.kt:17)
        at org.wordpress.android.localcontentmigration.TrustedQueryContentProvider.query(TrustedQueryContentProvider.kt:33)
        at android.content.ContentProvider.query(ContentProvider.java:1455)
        at android.content.ContentProvider.query(ContentProvider.java:1551)
        at android.content.ContentProvider$Transport.query(ContentProvider.java:285)
        at android.content.ContentResolver.query(ContentResolver.java:1219)
        at android.content.ContentResolver.query(ContentResolver.java:1151)
        at android.content.ContentResolver.query(ContentResolver.java:1107)
        at org.wordpress.android.localcontentmigration.LocalMigrationContentResolverKt.query(LocalMigrationContentResolver.kt:25)
        at org.wordpress.android.sharedlogin.resolver.LocalMigrationOrchestrator.tryLocalMigration(LocalMigrationOrchestrator.kt:161)
        at org.wordpress.android.ui.main.WPMainActivity.onCreate(WPMainActivity.java:463)
```

This seems unrelated with the changes of this PR but I'll investigate this further and provide more info :)

--- a/WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java
@@ -460,7 +460,10 @@ public class WPMainActivity extends LocaleAwareActivity implements
         if (!mSelectedSiteRepository.hasSelectedSite()) {
             initSelectedSite();
         }
-        mLocalMigrationOrchestrator.tryLocalMigration();
+
+        if (mBuildConfigWrapper.isJetpackApp()) {
+            mLocalMigrationOrchestrator.tryLocalMigration();
+        }
 
         // TODO: this is temporary to enable testing the migration flow UI
         if (mJetpackAppMigrationFlowUtils.shouldShowMigrationFlow()) {

